packages/ckeditor5-emoji/tests/emojimention.js

```diff
@@ -195,6 +196,163 @@ describe( 'EmojiMention', () => {
 		} );
 	} );
 
+	it( 'should set emoji mention feed configuration only once', async () => {
+		const editorElement = document.createElement( 'div' );
+		const editor1Element = document.createElement( 'div' );
+		document.body.appendChild( editorElement );
+		document.body.appendChild( editor1Element );
+
+		const editor = await ClassicTestEditor.create( editorElement, {
+			plugins: [ EmojiMention, Mention ],
+			substitutePlugins: [ EmojiDatabaseMock ]
+		} );
+
+		const editor1 = await ClassicTestEditor.create( editor1Element, {
+			plugins: [ EmojiMention, Mention ],
+			substitutePlugins: [ EmojiDatabaseMock ],
+			mention: {
+				feeds: editor.config.get( 'mention.feeds' )
+			}
+		} );
+
+		// Should register emoji mention config only once.
+		expect( editor1.config.get( 'mention.feeds' ).length ).to.equal( 1 );
+
+		await editor.destroy();
+		await editor1.destroy();
+		editorElement.remove();
+		editor1Element.remove();
+	} );
+
+	describe( 'should not update the mention configuration when emoji configuration is already added', () => {
+		let consoleWarnStub;
+
+		beforeEach( () => {
+			consoleWarnStub = sinon.stub( console, 'warn' );
+		} );
+
+		afterEach( () => {
+			consoleWarnStub.restore();
+		} );
+
+		it( 'should not call console warn when no additional mention or merge fields config is defined', async () => {
+			const editorElement = document.createElement( 'div' );
+			document.body.appendChild( editorElement );
+
+			const editor = await ClassicTestEditor.create( editorElement, {
+				plugins: [ EmojiMention, Mention ],
+				substitutePlugins: [ EmojiDatabaseMock ]
+			} );
+
+			expect( editor.config.get( 'mention.feeds' ).length ).to.equal( 1 );
+
+			editor.plugins.get( 'EmojiMention' )._setupMentionConfiguration( editor );
+
+			// Should not call console warn when there are no mention or merge fields configs defined.
+			expect( consoleWarnStub.callCount ).to.equal( 0 );
+			expect( editor.config.get( 'mention.feeds' ).length ).to.equal( 1 );
+
+			await editor.destroy();
+			editorElement.remove();
+		} );
+
+		it( 'should call console warn when merge fields config uses `:` prefix', async () => {
+			const editorElement = document.createElement( 'div' );
+			document.body.appendChild( editorElement );
+
+			const editor = await ClassicTestEditor.create( editorElement, {
+				plugins: [ EmojiMention, Mention ],
+				substitutePlugins: [ EmojiDatabaseMock ],
+				mergeFields: {
+					prefix: ':'
+				}
+			} );
+
+			expect( editor.config.get( 'mention.feeds' ).length ).to.equal( 0 );
+			expect( consoleWarnStub.callCount ).to.equal( 1 );
+
+			editor.plugins.get( 'EmojiMention' )._setupMentionConfiguration( editor );
```

Comment:
Calling `_setupMentionConfiguration()` to check if emoji config is not added again is already checked in test "should not call console warn when no additional mention or merge fields config is defined".

This `_setupMentionConfiguration()` is unnecessarily executed in these 3 new tests.

Reply:
Removed redundant tests in [3a50d9a](https://github.com/ckeditor/ckeditor5/pull/17826/commits/3a50d9af558b4047a3f132e651e37698a49e74e3)